src/app.rs

```diff
@@ -183,19 +307,116 @@ impl<'a, 'b, T: State + 'a> ApplicationBuilder<'a, 'b, T> {
         })
     }
 
-    /// Registers a given component type.
-    pub fn register<C: Component>(mut self) -> Self {
+    /// Registers a component into the entity-component-system. This method
+    /// takes no options other then the Component's type which is defined
```

Comment:
s/Component's/Components/

Reply:
I'm also fairly certain the Component owns its type.